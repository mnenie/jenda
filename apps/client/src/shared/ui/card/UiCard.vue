<script setup lang="ts">
defineSlots<{
  default: () => any
  user: () => any
}>()
</script>

<template>
  <div class="relative flex flex-col p-1.5 rounded-8px shadow-sm border border-solid border-neutral-200 text-sm dark:border-neutral-700">
    <slot />
    <div class="absolute bottom-1.5 top-1.5 w-fit">
      <slot name="user" />
    </div>
  </div>
</template>
