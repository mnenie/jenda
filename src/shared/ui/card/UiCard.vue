<script setup lang="ts">
defineSlots<{
  default: () => any;
  user: () => any;
}>();
</script>

<template>
  <div class="text-sm" :class="$style.card">
    <slot />
    <div :class="$style.user_container">
      <slot name="user" />
    </div>
  </div>
</template>

<style module lang="scss">
.card {
  position: relative;
  background-color: white;
  border: 1px solid var(--zinc-200);
  border-radius: 8px;
  box-shadow: 0 1px 2px 0 rgb(0 0 0 / 0.05);
  padding: 6px;
  width: 100%;
}

.user_container {
  position: absolute;
  bottom: 6px;
  right: 6px;
}
</style>
