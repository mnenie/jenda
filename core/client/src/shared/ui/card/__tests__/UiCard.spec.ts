import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import UiCard from "../UiCard.vue";

describe("uiCard", () => {
  const wrapper = mount(UiCard, {
    slots: {
      default: "card",
      user: `<div>users ðŸ‘¥</div>`,
    },
  });

  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
