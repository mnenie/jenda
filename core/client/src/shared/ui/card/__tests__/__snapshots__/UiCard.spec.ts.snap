// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiCard > should render correctly 1`] = `
"<div class="text-sm card">card<div class="user_container">
    <div>users ğŸ‘¥</div>
  </div>
</div>"
`;

exports[`uiCard > should render correctly 1`] = `
"<div class="text-sm card">card<div class="user_container">
    <div>users ğŸ‘¥</div>
  </div>
</div>"
`;
